functions/__tests__/config.test.ts

```diff
@@ -0,0 +1,137 @@
+import * as firebaseFunctionsTestInit from 'firebase-functions-test'
+import { readFileSync } from 'fs'
+import { resolve as pathResolve } from 'path'
+
+import * as yaml from 'js-yaml'
+import mockedEnv from 'mocked-env'
+
+let restoreEnv
+let extensionYaml
+let extensionParams
+let functionsConfig
+
+// Mocking of Firebase functions
+firebaseFunctionsTestInit()
+
+const environment = {
+  LOCATION: 'us-central1',
+  PROJECT_ID: 'fake-project',
+  COLLECTION_PATH: 'collection',
+  FIELDS_TO_INDEX: '',
+  SEARCHABLE_FIELDS: '',
+  MEILISEARCH_INDEX_NAME: 'example',
+  MEILISEARCH_HOST: 'http://127.0.0.1:7700',
+  MEILISEARCH_API_KEY: 'masterKey',
+}
+
+describe('extensions config', () => {
+  beforeAll(() => {
+    // Load the yaml documentation who contain all the environment parameters
+    extensionYaml = yaml.load(
+      readFileSync(pathResolve(__dirname, '../../extension.yaml'), 'utf8')
+    )
+
+    // Get all environment parameters in an object
+    extensionParams = extensionYaml.params.reduce((obj, param) => {
+      obj[param.param] = param
+      return obj
+    }, {})
+  })
+
```

Comment:
```suggestion

const EXTENSION_YAML = yaml.load(
  readFileSync(pathResolve(__dirname, '../../extension.yaml'), 'utf8')
)

const extensionParams = EXTENSION_YAML.params.reduce((obj, parameter) => {
  obj[parameter.param] = parameter
  return obj
}, {})

const environment = {
  LOCATION: 'us-central1',
  PROJECT_ID: 'fake-project',
  COLLECTION_PATH: 'collection',
  FIELDS_TO_INDEX: '',
  SEARCHABLE_FIELDS: '',
  MEILISEARCH_INDEX_NAME: 'example',
  MEILISEARCH_HOST: 'http://127.0.0.1:7700',
  MEILISEARCH_API_KEY: 'masterKey',
}

// Mocking of Firebase functions
firebaseFunctionsTestInit()

describe('extensions config', () => {
  let restoreEnv
  let functionsConfig

```

- moved `let` variables inside `describe` functions
- changed const variables: extensionYaml and extensionParams to the top and removed beforeAll
- Changed  a bit the reduce function as it was hard to read
- Changed position of `firebaseFuntionTestInit` to keep it close to its test context